mne/report.py

```diff
@@ -819,6 +848,11 @@ class Report(object):
         :meth:`mne.io.Raw.plot_psd`.
 
         .. versionadded:: 0.17
+    fname : str | None
+        Name of the file that is used as default when saving this report.
+        Defaults to ``None``, in which case it is saves as 'report.html' in
+        the current directory, or, if the :meth:`parse_folder` method was used,
+        as 'report.html' inside the parsed folder.
```

Comment:
I would just always use `cwd` and not relate it to `parse_folder`. I don't see much advantage to this contingent behavior. Users can also pass `parse_folder` more than once, then this behavior is not clear.

Also, why have `fname` at all? I'm not sure what the advantage is. If someone uses `Report` like in master (no context managing) it's not ever used. If `Report` is used with `open_report`, then `open_report` sets `report._fname = fname` after creation and it can be used then. What user-facing use case requires adding this param?

Reply:
alright, I'll make `fname` private.